common/persistence/sql/sqlMetadataManagerV2.go

```diff
@@ -171,6 +167,17 @@ func (m *sqlMetadataManagerV2) UpdateNamespace(request *persistence.InternalUpda
 	}
 
 	return m.txExecute("UpdateNamespace", func(tx sqlplugin.Tx) error {
+		metadata, err := lockMetadata(tx)
+		if err != nil {
+			return err
+		}
+		if metadata.NotificationVersion != request.NotificationVersion {
+			return fmt.Errorf(
```

Comment:
I don't understand when we return standart go errors and when `serviceerror`. I guess you following existing pattern here but we need to establish some rules here.

Reply:
yep, currently there is no proper translation of sql errors into modeled errors

one thing at a time.